torchx/lib/torchx/backend.ex

```diff
@@ -705,39 +705,107 @@ defmodule Torchx.Backend do
   end
 
   @impl true
-  def cumulative_sum(%T{type: out_type} = out, %T{} = t, axis) do
+  def cumulative_sum(%T{type: out_type} = out, %T{} = t, opts) do
     check_type!(out_type)
 
-    t
-    |> from_nx()
-    |> Torchx.cumulative_sum(axis)
-    |> to_nx(out)
+    axis = opts[:axis]
+    reverse = opts[:reverse]
+
+    t_tx =
+      if reverse do
+        t
+        |> from_nx()
+        |> Torchx.flip([axis])
+      else
+        from_nx(t)
+      end
+
+    result = Torchx.cumulative_sum(t_tx, axis)
+
+    if reverse do
+      result
+      |> Torchx.flip([axis])
+      |> to_nx(out)
+    else
+      to_nx(result, out)
+    end
   end
```

Comment:
We can refactor it into `cumulative_op(out, t, opts, &Torchx.cumulative_sum/2)` :)